lib/shared/src/sourcegraph-api/graphql/client.ts

```diff
@@ -1222,64 +1299,139 @@ export class SourcegraphGraphQLAPIClient {
+        }
+        this.cachedClientConfig = await this.refreshConfig()
+        return this.cachedClientConfig!
     }
 
     // Refreshes the config features by fetching them from the server and caching the result
-    public refreshConfigFeatures(): void {
-        const previousConfigFeatures = this.configFeatures
-        this.configFeatures = this.fetchConfigFeatures().catch((error: Error) => {
-            // Ignore fetcherrors as older SG instances will always face this because their GQL is outdated
-            if (!(error.message.includes('FetchError') || hasOutdatedAPIErrorMessages(error))) {
-                logError('ConfigFeaturesSingleton', 'refreshConfigFeatures', error.message)
-            }
-            // In case of an error, return previously fetched value
-            return previousConfigFeatures
-        })
+    public async refreshConfig(): Promise<CodyClientConfig> {
+        logDebug('ClientConfigSingleton', 'refreshing configuration')
+
+        // Determine based on the site version if /.api/client-config is available.
+        return graphqlClient
+            .getSiteVersion()
+            .then(siteVersion => {
+                if (isError(siteVersion)) {
+                    logError(
+                        'ClientConfigSingleton',
+                        'Failed to determine site version, GraphQL error',
+                        siteVersion
+                    )
+                    return false // assume /.api/client-config is not supported
+                }
+
+                // Insiders and dev builds support the new /.api/client-config endpoint
+                const insiderBuild = siteVersion.length > 12 || siteVersion.includes('dev')
+                if (insiderBuild) {
+                    return true
+                }
+
+                // Sourcegraph instances before 5.5.0 do not support the new /.api/client-config endpoint.
+                if (semver.lt(siteVersion, '5.5.0')) {
+                    return false
+                }
+                return true // Supports /.api/client-config
+            })
+            .then(supportsClientConfig => {
+                // If /.api/client-config is not available, fallback to the myriad of GraphQL
+                // requests that we previously used to determine the client configuration
+                if (!supportsClientConfig) {
+                    return this.fetchClientConfigLegacy()
+                }
+
+                // Otherwise we use our centralized client config endpoint.
+                return graphqlClient
+                    .fetchHTTP<CodyClientConfig>('client-config', 'GET', '/.api/client-config')
+                    .then(clientConfig => {
+                        if (isError(clientConfig)) {
+                            logError('ClientConfigSingleton', 'refresh client config', clientConfig)
+                            throw clientConfig
+                        }
+                        return clientConfig
```

Comment:
```suggestion
                        this.cachedClientConfig = clientConfig
                        return clientConfig
```

We gotta refresh the cache here